File: Codewars/python/7-kyu/4-solution.py
# Codewars 4
# Title : String ends with?
# Question : Complete the solution so that it returns true if the first argument(string) passed in ends with the 2nd argument (also a string). 
# Challange Url : https://www.codewars.com/kata/51f2d1cafc9c0f745c00037d
# Level : 7kyu


# Method : 1
def solution(text,ending):
    is_correct = False
    text_ending = text[(len(text) - len(ending)  ):len(text)]
    if ending == text_ending:
        is_correct = True
    return is_correct
    
# Method : 2
def solution(t,e): return True if t[(len(t) - len(e)):len(t)] == e else False
    
# Other user method
solution = str.endswith

def solution(string, ending):
    return ending in string[-len(ending):]

# Log or debug
print(solution("samurai", "ai" ))
print(solution("ninja",   "ja" ))
print(solution("sumo",    "omo"  ))

File: Codewars/typescript/7-kyu/4-solution.ts
// Codewars 4
// Title :  Sort Numbers
// Question :   Finish the solution so that it sorts the passed in array of numbers. If the function passes in an empty array or null/nil value then it should return an empty array.
// Challange Url :  https://www.codewars.com/kata/5174a4c0f2769dd8b1000003
// Level : 7kyu

// Method : 1
// export function solution(numbers: number[]): number[] {
//   for (let i = 0; i < numbers.length; i++) {
//     for (let j = 0; j < numbers.length - 1 - i; j++) {
//       if (numbers[j] > numbers[j + 1]) {
//         let temp = numbers[j];
//         numbers[j] = numbers[j + 1];
//         numbers[j + 1] = temp;
//       }
//     }
//   }

//   return numbers;
// }

// Method : 2
export const solution = (numbers: number[]): number[] =>
  numbers.sort((a, b) => a - b);

// Other Users Method
// export function solution(nums: number[]): number[] {
//   return nums ? nums.sort((a, b) => a - b) : [];
// }
